### PR TITLE
Feature/hide wki api

### PR DESCRIPTION
**Pull request by AshakaE**
# Query to hide wikis

_This endpoint is heavily guarded, it needs token validation and admin whitelist to execute query_

## How should this be tested?

1. image

## Notes or observations

_Unauthorized requests have a default response of `Forbidden resource`_

## Linked issues

closes a part of https://github.com/EveripediaNetwork/issues/issues/557

### CODE REVIEW

**Review by Softdev1 — APPROVED** [review]

File: src/App/utils/admin.guard.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { Observable } from 'rxjs'
+import validateToken from './validateToken'
+
+@Injectable()
+export default class AuthGuard implements CanActivate {
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+    // TODO: Make all admin ids lower case
+    const whitelist = ['0x5456afea3aa035088fe1f9aa36509b320360a89e']
```
Comment (kesar) [inline src/App/utils/admin.guard.ts:18]: plz add this list somewhere bcs its gonna be shared w other endpoints

Comment (AshakaE) [reply src/App/utils/admin.guard.ts:18]: Most definitely! What do you suggest for the api to organize homepage wikis? 🤔

Comment (kesar) [reply src/App/utils/admin.guard.ts:18]: i think wikis have a numeric value called promoted or similar, just an endpoint that you can pass a wikiId and a promoted number 👍🏻

File: src/App/wiki.resolver.ts [hunk 2]
```diff
@@ -44,6 +46,9 @@ class ByIdArgs {
 
   @Field(() => String)
   lang = 'en'
+
+  @Field(() => Int)
```
Comment (kesar) [inline src/App/wiki.resolver.ts:50]: extend that class and add this "number" also rename it to something that makes sense please, "number" does not reflect anything in particular.

File: src/App/wiki.resolver.ts [hunk 3]
```diff
@@ -95,6 +100,22 @@ class WikiResolver {
     })
   }
 
+  @Mutation(() => Wiki)
+  @UseGuards(AuthGuard)
+  async promoteWiki(@Args() args: ByIdArgs) {
+    const repository = this.connection.getRepository(Wiki)
+    const wiki = await repository.findOneOrFail(args.id)
+    if (wiki) {
```
Comment (kesar) [inline src/App/wiki.resolver.ts:108]: if its findOrFail you dont need a conditional here, it will fail before reaching this

**Review by kesar — APPROVED** [review]

File: src/App/wiki.resolver.ts [hunk 4]
```diff
@@ -133,6 +156,17 @@ class WikiResolver {
       .getMany()
   }
 
+  @Query(() => Wiki)
```
Comment (kesar) [inline src/App/wiki.resolver.ts:159]: ^- Mutation

Comment (AshakaE) [reply src/App/wiki.resolver.ts:159]: Oh snap 🤦‍♂️🤦‍♂️

**Pull request merged**